srml/session/src/lib.rs

```diff
@@ -190,11 +187,13 @@ impl<T: Trait> Module<T> {
 		T::OnSessionChange::on_session_change(time_elapsed, apply_rewards);
 
 		// Update any changes in session keys.
-		Self::validators().iter().enumerate().for_each(|(i, v)| {
-			if let Some(n) = <NextKeyFor<T>>::take(v) {
-				<consensus::Module<T>>::set_authority(i as u32, &n);
-			}
-		});
+		for (i, v) in Self::validators().into_iter().enumerate() {
+			<consensus::Module<T>>::set_authority(
+				i as u32,
+				&<NextKeyFor<T>>::get(&v)
+					.unwrap_or_else(|| T::ConvertAccountIdToSessionKey::convert(v))
```

Comment:
+ I assume this _default_ behavior is the most significant change right? 
+ does this have an associated issue?

Reply:
- Yes
- Sort of - #1829, however, the code isn't really broken, this just makes it cleaner.